### PR TITLE
I53 create mileage endpoint

### LINKED ISSUE #53: Create mileage endpoint
# Prerequisite
#40 Team API Endpoint

# Information
Create a new API endpoint for the "Mileage" db table using the Django REST framework. This task is split into multiple small parts that each team member can tackle! It should be under the folder server/api/mileage

1. Create Mileage Model in `models.py`
2. Create Views in `views.py`
It should have views for the following operations:
- create a new mileage entry
- get list of mileage by user_id
3.  Add url patterns to `urls.py`
4.  Register the app for the admin

You will be required to research how to complete this task!

TIP: You can use postman to test your endpoints!

Model data:
- mileage_id
- user_id
- kilometers
- date

### PR DESCRIPTION
**Pull request by jasperpato**
## Change Summary
Added mileage endpoint. Can POST mileage to a user and GET all mileages of a user. 

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [ ] The pull request title has an issue number
- [ ] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

# Related Issue

- Resolve #53

### CODE REVIEW

File: server/api/users/models.py
Comment (Dhruv8601) [inline server/api/users/models.py:1]: Should the FK be on_delete=models.CASCADE for optional params? I think this implies if the team is deleted, the user will also be deleted. Same thing for other optional params. 
I did it like this in events: team_id = models.ForeignKey(Team, on_delete=models.SET_NULL, null=True), so if the team is deleted, the team_id is just null for the event.

File: server/api/users/models.py [hunk 1]
```diff
@@ -55,8 +55,8 @@ class User(AbstractUser):
     team_signup = models.BooleanField(default=False)  # boolean
     has_consent = models.BooleanField(default=False)  # boolean
     travel_method = models.CharField(max_length=100, blank=True)
-    subteam_id = models.ForeignKey(SubTeam, null=True, on_delete=models.CASCADE)
-    team_id = models.ForeignKey(Team, null=True, on_delete=models.CASCADE)
+    subteam_id = models.ForeignKey(SubTeam, null=True, blank=True, on_delete=models.CASCADE)
+    team_id = models.ForeignKey(Team, null=True, blank=True, on_delete=models.CASCADE)
     team_admin = models.BooleanField(default=False)
```
Comment (Dhruv8601) [inline server/api/users/models.py:60]: Should the FK be on_delete=models.CASCADE for optional params? I think this implies if the team is deleted, the user will also be deleted. Same thing for other optional params. 
I did it like this in events: team_id = models.ForeignKey(Team, on_delete=models.SET_NULL, null=True), so if the team is deleted, the team_id is just null for the event.

Comment (jasperpato) [reply server/api/users/models.py:60]: Yes looks like it should definitely be SET_NULL. I didn't do this but I will push the change.

File: server/api/mileage/models.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from django.db import models
+import datetime
+from ..users.models import User
+
+
+class Mileage(models.Model):
+    mileage_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    kilometres = models.IntegerField()
```
Comment (ChantelleKerr) [inline server/api/mileage/models.py:9]: We should probably allow decimal values

**Review by ChantelleKerr — APPROVED** [review]
Great job, approved!

**Pull request merged**